src/main/java/com/epam/page/object/generator/validators/DuplicateTitleInInnerSearchRulesValidator.java

```diff
@@ -42,6 +50,6 @@ public boolean isValid(SearchRule searchRule, ValidationContext validationContex
 
     @Override
     public String getExceptionMessage(SearchRule searchRule, ValidationContext validationContext) {
-        return "Title = " + searchRule.getTitle() + " are duplicated";
```

Comment:
Search rule class is not an immutable class. If the value of title field changes, you will face an issue with inconsistency between the entity and "duplicate" field.

Reply:
@AleksandrKugushev  Good point!  We are going to replace SearchRule class with a hierarchy of different search rules (CommonSearchRule, ComplexSearchRule etc) during our refactoring. @FunCat  I suggest to leave it as is for now and think about immutability (or proper incapsulation) in the context of our new search rule subclasses